ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto

```diff
@@ -82,6 +83,10 @@ message CreatedEvent {
   // This means a newer client works with an older sandbox seamlessly.
   // Optional
   google.protobuf.StringValue agreement_text = 6;
+
+  // Metadata of the contract. Required for contracts created
```

Comment:
I actually would rather opt for an explicit flag on the ACS/Transaction streams and queries asking for this field to be populated. I would expect the metadata to be costly in some cases. And why carry it if no-one wants it?

Reply:
We will be already introducing flags for interface filters (see [here](https://github.com/digital-asset/daml/pull/14141/files#diff-bd749e6ba10cc644e517ea2592f798738ab24063cf525393bcc53247073e4bdbR65)) for whether to include interface views in the result, so we could do something similar.

> I would expect the metadata to be costly in some cases

Not sure about this part. The size of the metadata is constant (apart from maybe the driver metadata on Canton ledgers) and doesn't require any computation. With the flag, you could easily save on bandwidth between the participant and client application (is that relevant?). You could also save on database resources by not including the key hash and driver metadata columns in database queries, but then you would have to wire up the flag all the way to where SQL queries are generated.

Reply:
I have deferred the decision on this implementation for two reasons:
1. The current high-performance ledgers do not forward driverMetadata, which, as Robert mentioned, is the only unknown in terms of size. I agree with the other aspect as well: the possibility to toggle keyHash on or off is going to lead to higher code complexity with marginal benefit.
2. The estimates on the size in Canton is still an unknown since it is dependent on future work (see discussion [here](https://github.com/digital-asset/daml/pull/13846#discussion_r913533988))

Reply:
We just got an statement about the size of driver metadata on canton: it is likely unbounded (https://github.com/digital-asset/daml/pull/13846#discussion_r913533988)

Reply:
How about adding a story to gauge performance impact using an arbitrary metadata record of XXX bytes and reacting to that?

Reply:
Let's then change the comment to state that this field will be empty until the explicit disclosure feature is turned on

Reply:
Done c85c31009997896988fa583cc1ad7fc3b02c3c52